### PR TITLE
Web Apps: switched combobox to select2

### PR DESCRIPTION
**Pull request by orangejenny**
##### SUMMARY
https://dimagi-dev.atlassian.net/browse/USH-394

The atwho widget used for comboboxes is not at all accessible to screen readers. atwho is [no longer maintained](https://github.com/ichord/At.js/#%EF%B8%8F-announcement-%EF%B8%8F), so that's not going to improve.

select2's upcoming release has [a number of accessibility improvements](https://github.com/select2/select2/issues/3744), although it isn't yet released and there's [some debate](https://github.com/woocommerce/selectWoo/issues/35) about whether those improvements will be sufficient. My hope is that switching over to select2 will be enough of an improvement for the functionality needed in web apps, which is pretty basic. At a minimum, select2 is at least backed by a select element instead of a text box. We'll be testing this from an accessibility perspective and can consider additional changes, possibly switching to [selectWoo](https://developer.woocommerce.com/2017/08/08/selectwoo-an-accessible-replacement-for-select2/), if necessary.

This PR also updates "minimal" select questions, which were previously using plain HTML select elements, to use select2 for the sake of consistency. This makes the minimal and combobox options on web apps very similar, with the only difference being that the combobox allows for selecting the specific matching algorithm, whereas minimal uses select2's default search.

##### RISK ASSESSMENT / QA PLAN
[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-2058)

##### PRODUCT DESCRIPTION
Makes fairly minor changes to the appearance both both comboboxes and "minimal" select questions in web apps.

Before, comboboxes use atwho and minimal questions use standard HTML dropdowns:
Screen Shot 2020-10-28 at 12 36 52 PM

Screen Shot 2020-10-28 at 12 37 21 PM

After, both widgets use select2:
Screen Shot 2020-10-28 at 12 39 10 PM

Screen Shot 2020-10-28 at 12 24 31 PM

### CODE REVIEW

**Comment by orangejenny** [comment]
FYI @dimagi/product

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 1]
```diff
@@ -530,15 +530,39 @@ hqDefine("cloudcare/js/form_entry/entrycontrols_full", function () {
...
         EntrySingleAnswer.call(this, question, options);
         self.templateType = 'dropdown';
+        self.placeholderText = gettext('Please choose an item');
 
-        self.options = ko.pureComputed(function () {
+        self.helpText = function () {
+            return "";
+        };
+
+        self.options = ko.computed(function () {
```
Comment (biyeun) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js:539]: i'm guessing `pureComputed` doesn't have the `subscribe` option?

**Review by biyeun — APPROVED** [review]
this is looking great. is accessibility part of a USH initiative / is there some plan to looking into moving away from `atwho` on `vellum`? I'm sad another useful js library has now fallen into the unmaintained category. 😞

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js:539]: It does, but the subscribers shouldn't have side effects. I think that rendering the select2 counts as a side effect, although I'm totally certain. I'm looking at the "When not to use a pure computed observable" section in these docs: https://knockoutjs.com/documentation/computed-pure.html

**Comment by orangejenny** [comment]
@biyeun Agreed, I like atwho. Although some of the places we use it, like this one, the underlying concept is a dropdown and we only used atwho because the UI is lighter-weight than the old version of select2. In these places, I'm happy to replace with new select2. I think that applies to vellum's usages of atwho. Places where atwho is really being used as it's intended, like in the web apps debugger where it autocompletes function names in xpath expressions, are trickier.

As far as planning/resourcing, USH is indeed working on accessibility, but we're focused on web apps usage, so we're unlikely to update vellum anytime soon. Let's offline if you want to chat more about this.

**Review by biyeun — APPROVED** [review]
🥳

**Comment by orangejenny** [comment]
@biyeun Requesting review once more, please - the last few commits changed validation logic a bit.

**Pull request merged**